os/linux_base.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * @APPLE_APACHE_LICENSE_HEADER_START@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @APPLE_APACHE_LICENSE_HEADER_END@
+ */
+
+#ifndef __OS_LINUX_BASE__
+#define __OS_LINUX_BASE__
+
+
+// marker for hacks we have made to make progress
+#define __LINUX_PORT_HDD__ 1 
+
+/*
+ * Stub out defines for some mach types and related macros
+ */
+
+typedef uint32_t mach_port_t;
+
+#define  MACH_PORT_NULL 0
+#define  MACH_PORT_DEAD -1
+
+#define EVFILT_MACHPORT -8
+
+typedef uint32_t mach_error_t;
+
+typedef uint32_t mach_vm_size_t;
+
+typedef uint32_t mach_msg_return_t;
+
+typedef uintptr_t mach_vm_address_t;
+
+typedef uint32_t dispatch_mach_msg_t;
+
+typedef uint32_t dispatch_mach_t;
+
+typedef uint32_t dispatch_mach_reason_t;
+
+typedef uint32_t voucher_activity_mode_t;
+
+typedef uint32_t voucher_activity_trace_id_t;
+
+typedef uint32_t voucher_activity_id_t;
+
+typedef uint32_t _voucher_activity_buffer_hook_t;;
+
+typedef uint32_t voucher_activity_flag_t;
+
+typedef struct
+{
+} mach_msg_header_t;
+
+
+typedef void (*dispatch_mach_handler_function_t)(void*, dispatch_mach_reason_t,
+						 dispatch_mach_msg_t, mach_error_t);
+
+typedef void (*dispatch_mach_msg_destructor_t)(void*);
+
+/*
+ * Stub out defines for other missing types
+ */
+
+struct kevent64_s
+{
+    uint64_t        ident;
+    int16_t         filter;
+    uint16_t        flags;
+    uint32_t        fflags;
+    int64_t         data;
+    uint64_t        udata;
+    uint64_t        ext[2];
+};
+
+typedef uint32_t voucher_activity_mode_t;
+
+struct voucher_offsets_s {
+  uint32_t vo_version;
+};
+
+
+// bogus...
+#define PAGE_SIZE 4096
+#define SIZE_T_MAX 0x7fffffff
+
+#define NOTE_VM_PRESSURE 0
+#define NOTE_ABSOLUTE 0
+#define NOTE_NSECONDS 0
+#define NOTE_LEEWAY 0
+#define NOTE_CRITICAL 0
+#define NOTE_BACKGROUND 0
+
+/*
+ * Stub out misc linking and compilation attributes
+ */
+
+#ifdef OS_EXPORT
```

Comment:
it would be better to put the `OS_*` vmacros from the real os/base.h into the replicated version in the libdispatch repo (if we go that route)